src/guiFormSpecMenu.cpp

```diff
@@ -636,17 +636,17 @@ void GUIFormSpecMenu::parseBackground(parserData* data,std::string element)
 		geom.X = stof(v_geom[0]) * (float)spacing.X;
 		geom.Y = stof(v_geom[1]) * (float)spacing.Y;
 
-		if (parts.size() == 4) {
-			m_clipbackground = is_yes(parts[3]);
-			if (m_clipbackground) {
-				pos.X = stoi(v_pos[0]); //acts as offset
-				pos.Y = stoi(v_pos[1]); //acts as offset
-			}
+		if (!data->explicit_size)
+			warningstream<<"invalid use of background without a size[] element"<<std::endl;
+
+		if (parts.size() == 4 && is_yes(parts[3])) {
```

Comment:
In order to have better self-documenting code it would be cool to have something more like:
```c++
bool clip = false;
if (parts.size() == 4 && is_yes(parts[3])) {
	// etc etc
	clip = true;
}
m_backgrounds.push_back(ImageDrawSpec(name, pos, geom, clip));
```